admin/resources/js/components/UpdateUser.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import {
+  Language,
+  UpdateUserInput,
+  User,
+  useUpdateUserMutation,
+} from "../api/generated";
+import errorMessages from "./form/errorMessages";
+import Input from "./form/Input";
+import Select from "./form/Select";
+import Submit from "./form/Submit";
+
+type FormValues = UpdateUserInput;
+interface UpdateUserFormProps {
+  initialUser: User;
+  handleUpdateUser: (id: string, data: FormValues) => Promise<FormValues>;
+}
+
+export const UpdateUserForm: React.FunctionComponent<UpdateUserFormProps> = ({
+  initialUser,
+  handleUpdateUser,
+}) => {
+  const methods = useForm<FormValues>({ defaultValues: initialUser });
+  const { handleSubmit, reset } = methods;
+
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    return handleUpdateUser(initialUser.id, data)
+      .then(reset) // Reset form with returned data. This resets isDirty flag.
+      .catch(() => {
+        // Something went wrong with handleUserUpdate.
+        // Do nothing.
+      });
+  };
+
+  return (
+    <section>
+      <h2>Update a User</h2>
+      <FormProvider {...methods}>
+        <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
Its because I wanted to call reset, from useForm, inside onSubmit.
It's possible to update `<Form />` to allow this, by having it accept a handleSubmit function which takes a second formContext argument (eg `onSubmit = (data, {reset}) => ...`) but then `<Form />` started to look more complicated and I wasn't sure it's worth it.

Reply:
Hmm, okay that makes sense for now. There might be a way around it, but for now this is fine.